Make rebind<U>::other explicit in FastAllocator

rkirsling opened the pull request:
#### 4e1abb6514cdc2bf301e5ea4f299c139faa0b2ff
<pre>
Make rebind&lt;U&gt;::other explicit in FastAllocator
<a href="https://bugs.webkit.org/show_bug.cgi?id=265933">https://bugs.webkit.org/show_bug.cgi?id=265933</a>

Reviewed by Yusuke Suzuki.

PlayStation build fix following 271419@main.
Evidently using StdUnorderedMap&lt;K,V&gt;::extract(K) causes our platform to complain about rebind&lt;U&gt;::other being undefined.
It&apos;s trivial to define so let&apos;s just do so.

* Source/WTF/wtf/FastMalloc.h:

Canonical link: <a href="https://commits.webkit.org/271714@main">https://commits.webkit.org/271714@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dcb60f91d758d0695e065cd2ea296b4137357513

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28970 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7638 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30326 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31584 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26395 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9758 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4958 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26419 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29242 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6291 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24857 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5482 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5603 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25882 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/32922 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/25013 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26477 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26317 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31864 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/29242 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5585 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3758 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29645 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7237 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/35585 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6990 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6078 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7672 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6106 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
Hmm, looks like the platform's library is broken since it should not be necessary if `std::allocator_traits` is correctly used. But r=me for now.
See https://en.cppreference.com/w/cpp/memory/allocator_traits

rkirsling committed:
Make rebind<U>::other explicit in FastAllocator
https://bugs.webkit.org/show_bug.cgi?id=265933

Reviewed by Yusuke Suzuki.

PlayStation build fix following 271419@main.
Evidently using StdUnorderedMap<K,V>::extract(K) causes our platform to complain about rebind<U>::other being undefined.
It's trivial to define so let's just do so.

* Source/WTF/wtf/FastMalloc.h:

Canonical link: https://commits.webkit.org/271714@main
